Attributes/Troops/Logic/attribute_5.gd

```diff
@@ -0,0 +1,29 @@
+extends TroopAttribute
+
+# Logic for splash attribute
+
+func build_action() -> Action:
+    var action: Action = Action.new()
+    action.name = attribute.name
+    action.description = attribute.description
+    action.setup(parent.game, on_attack)
+    return action
+
+#loops to all tiles around defender and applies half damage
+
+func on_attack(defender: Unit):
+    var defender_pos = defender.pos
+    var splash_tiles = _get_surrounding(defender_pos, 1)
+
+    for tile in splash_tiles:
+        var unit_at_tile = game.board.units[tile.x][tile.y]
+        apply_splash_damage(unit_at_tile)
+
+#Applies half attack damage to enemy
+
+func apply_splash_damage(troop: Troop):
+    var splash_damage = floor(base_stats.attack / 2.0)
```

Comment:
Take a look at the damage calculations on lines 242, 224, 222 [here](https://github.com/Open-Source-Software-Development/Cardtorium/blob/main/Classes/troop.gd) and change what's here to match. Currently you're using the defending unit's attack, but you should also pass in the attacking unit and use it's attack.